marshal.go

```diff
@@ -33,12 +34,12 @@ func extractInt(n *Integer, m map[string]interface{}, s string) error {
 
 	switch v.(type) {
```

Comment:
This is probably better written as 
```go
switch val := v.(type) {
case float64:
    n.Val = int(val)
    n.Initialized = true
    return nil
default:
    return errors.Wrap(errInvalidType("float64", v), "failed to extract int")
}
```
the val := v.(type) makes a new variable of the appropriate type, so you don't have to re-convert.

Or, since there's only ever one expected type, you can just do this:

```go
val, ok := v.(float64)
if !ok {
    return errors.Wrap(errInvalidType("float64", v), "failed to extract int")
}
n.Val = int(val)
n.Initialized = true
return nil
```